torch/_inductor/compile_fx.py

```diff
@@ -2362,3 +2368,136 @@ def _aoti_flatten_inputs(
         }
     )
     return flat_example_inputs, options
+
+
+_ENCODING_VERSION: int = 1
+
+
+class CompiledArtifact:
+    _compiled_fn: Callable[..., Any]
+    _artifacts: Optional[tuple[bytes, CacheInfo]]
+
+    def __init__(
+        self,
+        compiled_fn: Callable[..., Any],
+        artifacts: Optional[tuple[bytes, CacheInfo]],
+    ):
+        self._compiled_fn = compiled_fn
+        self._artifacts = artifacts
+
+    def __call__(self, *args: Any) -> Any:
+        return self._compiled_fn(*args)[0]
+
+    def save(
+        self, *, path: str, format: Literal["binary", "unpacked"] = "binary"
+    ) -> None:
+        with dynamo_timed("CompiledArtifact.save"):
+            if self._artifacts is None:
+                raise RuntimeError(
+                    "CompiledArtifact.save failed to save since there's no artifact to save"
+                )
+            artifact_bytes, cache_info = self._artifacts
+            assert len(cache_info.aot_autograd_artifacts) == 1
+            key = cache_info.aot_autograd_artifacts[0]
+
+            if format == "binary":
+                # cant assert that it is a file since it might not exist yet
+                assert not os.path.isdir(path)
+
+                from torch.utils._appending_byte_serializer import BytesWriter
+
+                writer = BytesWriter(0)
+                writer.write_uint64(_ENCODING_VERSION)
+                writer.write_str(key)
+                writer.write_bytes(artifact_bytes)
+                with open(path, "wb") as file:
+                    file.write(writer.to_bytes())
+            else:
+                assert format == "unpacked"
+                assert os.path.isdir(path)
```

Comment:
We probably should do a rm on the directory first? It's like how in the binary path the writer overwrites the file (does it?)